arangod/Agency/Agent.cpp

```diff
@@ -1664,12 +1663,18 @@ void Agent::executeLockedWrite(std::function<void()> const& cb) {
   cb();
 }
 
+void Agent::executeTransientLocked(std::function<void()> const& cb) {
+  _tiLock.assertNotLockedByCurrentThread();
```

Comment:
Is this still necessary? It seems to me that it's not. And if it is, _tiLock should be documented accordingly.